web-api/src/ingestion/routes.rs

```diff
@@ -134,50 +152,77 @@ async fn new_documents(
 
     let start = Instant::now();
 
-    let (documents, mut failed_documents) = body
+    let (preprocessed, mut failed_documents): (Vec<_>, Vec<_>) = body
         .documents
         .into_iter()
-        .map(|document| {
-            let map_document = || -> anyhow::Result<_> {
-                let id = document.id.as_str().try_into()?;
-                let properties = document
-                    .properties
-                    .into_iter()
-                    .map(|(id, property)| id.try_into().map(|id| (id, property)))
-                    .try_collect()?;
-                let tags = document
-                    .tags
-                    .into_iter()
-                    .map(TryInto::try_into)
-                    .try_collect()?;
-                let embedding = state.embedder.run(&document.snippet)?;
-
-                Ok(models::IngestedDocument {
-                    id,
-                    snippet: document.snippet,
-                    properties,
-                    tags,
-                    embedding,
-                    is_candidate: document.is_candidate,
+        .map(
+            |IngestedDocument {
+                 id,
+                 snippet,
+                 properties,
+                 tags,
+                 is_candidate,
+                 default_is_candidate,
+             }| {
+                let map_document = || -> anyhow::Result<_> {
+                    let id = id.as_str().try_into()?;
+                    let properties = properties
+                        .into_iter()
+                        .map(|(id, property)| id.try_into().map(|id| (id, property)))
+                        .try_collect()?;
+                    let tags = tags.into_iter().map(TryInto::try_into).try_collect()?;
+                    let is_candidate_op = match (is_candidate, default_is_candidate) {
+                        (Some(_), Some(_)) => {
+                            bail!("You can only use either of is_candidate or default_is_candidate")
+                        }
+                        (Some(value), None) => IsCandidateOp::SetTo(value),
+                        (None, Some(value)) => IsCandidateOp::DefaultTo(value),
+                        (None, None) => IsCandidateOp::SetTo(true),
+                    };
+                    //FIXME fetch snippet with is_candidate below and only calculate/set
+                    //      the embedding if the snippet changed.
```

Comment:
:art: this is part of [ET-4237], not sure if we need an extra fixme comment here.